orion-server/src/main/java/com/pinterest/orion/core/actions/kafka/ClusterRecoveryAction.java

```diff
@@ -0,0 +1,223 @@
+            logger.info(restartNote);
+            getResult().appendOut(restartNote);
+        }
+        String dispatchNote = String.format(
+                "Dispatching BrokerRecoveryAction for node %s in cluster %s. ",
+                deadBrokerId,
+                cluster.getClusterId());
+        logger.info(dispatchNote);
+        getResult().appendOut(dispatchNote);
+        getEngine().dispatch(brokerRecoveryAction);
+    }
+
+    protected boolean healBrokers(Set<String> candidates) throws Exception {
+        String output;
+        boolean isSucceed = false;
+        if (candidates.size() == 1) {
+            output = String.format(
+                    "ClusterRecoveryAction trys to recover broker %s. ",
+                    candidates.iterator().next());
+            String deadBrokerId = candidates.iterator().next();
+            healBroker(deadBrokerId);
+            logger.info(output);
+            isSucceed = true;
+        } else if (candidates.size() > 1){
+            // more than 1 brokers are dead... better alert and have human intervention
+            output = String.format("More than one brokers are in bad state - dead: %s, service down: %s. " +
+                            "ClusterRecoveryAction skips automatic recovery and pages oncall. ",
+                    deadBrokers, maybeDeadBrokers);
+            cluster.getActionEngine().alert(AlertLevel.HIGH, new AlertMessage(
+                    candidates.size() + " brokers on " + cluster.getClusterId() + " are unhealthy",
+                    output,
+                    "orion"
+            ));
+            OrionServer.metricsCounterInc(
+                    "broker.services.unhealthy",
+                    new HashMap<String, String>() {{
+                        put("clusterId", cluster.getClusterId());
+                    }}
+            );
+            logger.severe(output);
+        } else {
+            output = String.format("No candidates for healing in cluster %s. ", cluster.getClusterId());
+            logger.warning(output);
+        }
+        getResult().appendOut(output);
+        return isSucceed;
+    }
+
+    public void setCandidates(Set<String> candidates) {
+        this.candidates = candidates;
+    }
+
+    public void setDeadBrokers(Set<String> deadBrokers) {
+        this.deadBrokers = deadBrokers;
+    }
+
+    public void setMaybeDeadBrokers(Set<String> maybeDeadBrokers) {
+        this.maybeDeadBrokers = maybeDeadBrokers;
+    }
+
+    public void setNonExistentBrokers(Set<String> nonExistentBrokers) {
+        this.nonExistentBrokers = nonExistentBrokers;
+    }
+
+    public void setCluster(Cluster cluster) {
+        this.cluster = cluster;
+    }
+
+    public void setSensorSet(Set<String> sensorSet) {
+        this.sensorSet = sensorSet;
+    }
+
+    protected static String deriveNonexistentHostname(String existingHostname, String existingId, String nonExistingId) {
```

Comment:
Please document the function and its inputs/output using javadoc.

Reply:
This is legancy code copying from BrokerHealingOperator. Will add doc here.